synchronization/trigger_test.go

```diff
@@ -80,10 +80,10 @@ func TestPeriodicalTriggerStopOnContextCancel(t *testing.T) {
 func TestPeriodicalTriggerStopWorksWhenContextIsCancelled(t *testing.T) {
 	with.Concurrency(t, func(parent context.Context, harness *with.ConcurrencyHarness) {
 		ctx, cancel := context.WithCancel(parent)
-		cancel() // send a cancelled context to reduce chances of trigger being called even once
 		x := 0
 		p := synchronization.NewPeriodicalTrigger(ctx, "a periodical trigger", time.Millisecond*2, harness.Logger, func() { x++ }, nil)
 		harness.Supervise(p)
+		cancel()
```

Comment:
See the comment above. The reason we send a cancelled context is that the test was flaky to begin with. This change will make it flaky in a different manner. Maybe change the timings or choose another strategy to assert that it doesn't flake?